### PR TITLE
refactor: Simplify some internals of `qiskit_to_tk` conversion

### PR DESCRIPTION
**Pull request by CalMacCQ**
# Description

This PR begins a refactor of the `qiskit_to_tk` and `tk_to_qiskit` converters. Here I've focussed on refactoring some of the large `CircuitBuilder.add_qiskit_data` method (used as part of `qiskit_to_tk` into separate internal functions. I've added some short docstrings so its hopefully clearer whats going on.

There's still a lot to improve in the conversion code but I think it makes sense to do this refactor in several stages to make small fixes like https://github.com/CQCL/pytket-qiskit/issues/366 a bit easier.

I've avoided imporvements which involve user facing changes for now. Will add these in a follow up.

# Related issues

#313 

# Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented hard-to-understand parts of my code.
- [ ] I have made corresponding changes to the public API documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have updated the changelog with any user-facing changes.

### CODE REVIEW

File: pytket/extensions/qiskit/qiskit_convert.py [hunk 1]
```diff
@@ -550,15 +579,12 @@ def qiskit_to_tk(qcirc: QuantumCircuit, preserve_param_uuid: bool = False) -> Ci
     Converts a qiskit :py:class:`qiskit.QuantumCircuit` to a pytket :py:class:`Circuit`.
 
     :param qcirc: A circuit to be converted
-    :type qcirc: QuantumCircuit
```
Comment (CalMacCQ) [inline pytket/extensions/qiskit/qiskit_convert.py:553]: These type declarations in the docstrings are redundant as the types are inferred from the function/class signiture. This inference is done by the [sphinx-autodoc-typehints package](https://github.com/tox-dev/sphinx-autodoc-typehints) package which is used in the docs build.

**Comment by CalMacCQ** [comment]
In fact I think its better to do the typing updates in a separate PR.

File: pytket/extensions/qiskit/qiskit_convert.py [hunk 2]
```diff
@@ -247,43 +248,157 @@ def _qpo_from_peg(peg: PauliEvolutionGate, qubits: list[Qubit]) -> QubitPauliOpe
...
+    else:
+        if isinstance(prep.params[0], complex):
+            # convert int to a binary string and apply X for |1>
+            integer_parameter = int(prep.params[0].real)
+            bit_string = bin(integer_parameter)[2:]
+            circuit = _string_to_circuit(bit_string, prep.num_qubits, qiskit_prep=prep)
+            tkc.add_circuit(circuit, qubits)
+        else:
+            raise TypeError(
+                "Unrecognised type of Instruction.params "
```
Comment (CalMacCQ) [inline pytket/extensions/qiskit/qiskit_convert.py:371]: I'm not so satisfied with this `_add_state_preparataion` function. There are 3 different ways to create a `StatePreparation` or `Initialize` object. Couldn't simplify the handling all that much.

File: pytket/extensions/qiskit/qiskit_convert.py [hunk 3]
```diff
@@ -247,43 +248,157 @@ def _qpo_from_peg(peg: PauliEvolutionGate, qubits: list[Qubit]) -> QubitPauliOpe
...
+    return circ
+
+
+def _get_controlled_tket_optype(c_gate: ControlledGate) -> OpType:
+    """Get a pytket contolled OpType from a qiskit ControlledGate."""
+    if c_gate.base_class in _known_qiskit_gate:
+        # First we check if the gate is in _known_qiskit_gate
+        # this avoids CZ being converted to CnZ
+        known_optype = _known_qiskit_gate[c_gate.base_class]
+        return known_optype
```
Comment (cqc-alec) [inline pytket/extensions/qiskit/qiskit_convert.py:299]:
```suggestion
        return _known_qiskit_gate[c_gate.base_class]
```

File: pytket/extensions/qiskit/qiskit_convert.py [hunk 4]
```diff
@@ -247,43 +248,157 @@ def _qpo_from_peg(peg: PauliEvolutionGate, qubits: list[Qubit]) -> QubitPauliOpe
...
+            reversed_qubits = list(reversed(qubits))
+            tkc.add_gate(pytket_state_prep_box, reversed_qubits)
+    else:
+        if isinstance(prep.params[0], complex):
+            # convert int to a binary string and apply X for |1>
+            integer_parameter = int(prep.params[0].real)
+            bit_string = bin(integer_parameter)[2:]
+            circuit = _string_to_circuit(bit_string, prep.num_qubits, qiskit_prep=prep)
+            tkc.add_circuit(circuit, qubits)
+        else:
```
Comment (cqc-alec) [inline pytket/extensions/qiskit/qiskit_convert.py:369]: Use `elif` and `else` here to save indentation?

**Review by cqc-alec — COMMENTED** [review]
Thanks, just a couple of minor suggestions.

File: pytket/extensions/qiskit/qiskit_convert.py (hunk 4 above)
Comment (CalMacCQ) [reply pytket/extensions/qiskit/qiskit_convert.py]: Thanks, done in [37de5e4](https://github.com/CQCL/pytket-qiskit/pull/382/commits/37de5e465c6d2e715399ffd34fb8663f8216c9ee)

File: pytket/extensions/qiskit/qiskit_convert.py (hunk 3 above)
Comment (CalMacCQ) [reply pytket/extensions/qiskit/qiskit_convert.py]: Done in [37de5e4](https://github.com/CQCL/pytket-qiskit/pull/382/commits/37de5e465c6d2e715399ffd34fb8663f8216c9ee)

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**